docs/Configuration-Settings.md
## Taranis

Press the `Menu` button (`Shift` on X-Lite) to display the configuration options menu:

* Use the dial or +/- buttons to cycle through the menu or select the desired setting
* Press Enter/dial to select and deselect a menu option
* Press `Exit` or `RTN` to deselect a menu option or escape the configuration menu

## Horus/Jumper T16

* Must be unarmed to access the config menu when running as a widget
* Use the following (mode 2) stick controls to navigate the config menu:

![sample](https://raw.githubusercontent.com/iNavFlight/LuaTelemetry/master/assets/iNavConfigHorus.png "Horus config menu")

### Configuration menu options

![sample](https://raw.githubusercontent.com/iNavFlight/LuaTelemetry/master/assets/iNavConfig.png "Configuration menu")

!!! note
    Options without selections have either been disabled by another option setting or are not available on your transmitter or telemetry protocol.

  * **Battery View** - Total battery voltage / Cell voltage average (default: Total)
  * **Battery Alert** - All battery alerts on, off or only critical alerts (default: All)
  * **Cell Low** - Cell voltage for low battery warning (default: 3.5V) **[[help](#suggested-battery-settings)]**
  * **Cell Critical** - Cell voltage for battery critical warning (default: 3.4V) **[[help](#suggested-battery-settings)]**
  * **Cell Calculation** - Voltage to calculate cell count, 4.3V for LiPo, 4.4V for LiHV (default: 4.3V)
  * **Fuel Unit** - Match to INAV CLI value `smartport_fuel_unit` (default: Percent) **[[help](#suggested-battery-settings)]**
  * **Battery Capacity** - Set the battery capacity for Crossfire (default: 1500mAh)
  * **Fuel Low** - Fuel percentage for low battery warning (default: 30%) **[[help](#suggested-battery-settings)]**
  * **Fuel Critical** - Fuel percentage for battery critical warning (default: 20%) **[[help](#suggested-battery-settings)]**
  * **Altitude Alert** - Turn on/off the altitude alert (default: On)
  * **Max Altitude** - Altitude warning starts when over this value (default: 400' or 120m)
  * **Timer** - Show the automatic flight timer, timer1, timer2 or turn timer off (default: Auto)
  * **Tx Voltage** - Show transmitter voltage as graph and/or numerical value (default: Both or Graph)
  * **Rx Voltage** - Turn on/off the receiver voltage in the title (default: On)
  * **Variometer** - Vertical speed graph, voice notifications or off (default: Off)
  * **Vario Steps** - If Variometer is set to Voice, select the altitude steps (default: 10' or 10m)
  * **Altitude Graph** - Turn off or set altitude graph time period from 1-6 minutes (default: Off)
  * **Voice Alerts** - All voice alerts on, off or only critical alerts (default: All)
  * **Feedback** - Turn beeper and/or haptic feedback for alerts on or off (default: All)
  * **RTH Feedback** - Return to home beeper and haptic feedback on or off (default: On)
  * **HeadFree Feedback** - Head free beeper and haptic feedback on or off (default: On)
  * **RSSI Feedback** - RSSI beeper and haptic feedback on or off (default: On)
  * **AltHold Center FB** - Hepatic/audio feedback for Alt Hold center throttle position (default: Off)
  * **Speed Sensor** - Speed sensor to use, GPS or (if available) Pitot air speed (default: GPS)
  * **View Mode** - Classic, pilot, radar or altitude graph view modes (default: Classic)
  * **Roll Indicator** - Turn on/off roll indicator on Horus (default: Off)
  * **Aircraft Symbol** - Options 0 - 5 on Horus, [see below](#horusjumper-t16-aircraft-symbol-options) for options (default: 0)
  * **HUD Home Icon** - Select HUD or simple orientation for home icon on Horus (default: Off)
  * **Center Map Home** - Turn on/off centering of the radar home position on Horus (default: Off)
  * **Orientation** - Launch/pilot-based or compass-based default orientation (default: Launch)
  * **GPS HDOP View** - View the GPS accuracy (HDOP) as a Graph or Decimal (default: Graph)
  * **GPS Warning** - GPS accuracy (HDOP) to trigger warning (default: > 3.5 HDOP [at least 1 bar])
  * **GPS** - GPS coordinates as decimal or degrees/minutes format (default: Decimal)
  * **Playback Log** - Playback telemetry log files (latest 5 logs from the last 2 weeks) **[[help](../Configuration-Settings/#playback-telemetry-log-files)]**

## Suggested Battery Settings
### Voltage and Current Calibration

1. Using a multimeter, calibrate the voltage with the "Voltage Scale" in INAV configurator
1. If you have a current sensor, make sure you [calibrate it](https://www.youtube.com/watch?v=AWjblvHgjjI)

### In INAV Configurator

* Voltage source to use for alarms and telemetry: **Raw**
* Number of cells: **0** (0=auto, set if you always use the same cell count)
* Maximum cell voltage for cell count detection: **4.3**
* Minimum cell voltage: **3.4** (match "Cell Critical" in Lua Telemetry)
* Maximum cell voltage: **4.2**
* Warning cell voltage: **3.5** (match "Cell Low" in Lua Telemetry)
* Set capacity in **mAh** to battery capacity
* Warning capacity: **30%** (match "Fuel Low" in Lua Telemetry)
* Critical capacity: **10%**

### In INAV CLI (ignore for TBS Crossfire)

* `set smartport_fuel_unit = percent`
* `set report_cell_voltage = OFF` (if set to `ON` Lua Telemetry can't show total battery voltage)

### In Lua Telemetry

* Cell Low: **3.5V** (match "Warning cell voltage" in INAV)
* Cell Critical: **3.4V** (match "Minimum cell voltage" in INAV)
* Fuel Unit: **Percent**
* Fuel Low: **30%** (match "Warning capacity" in INAV)
* Fuel Critical: **20%**

## Playback Telemetry Log Files

!!!warning
    Telemetry log file playback works really well on Horus transmitters.  However, while it works on my Taranis Q X7, for others it may crash with a "**attempt to call a nil value**" or "**not enough memory**" error.  This means you're using too much memory for other things on your transmitter and it doesn't have enough memory to playback the logs.  Not switching views and not using the pilot view will use less memory, but it's possible you just won't be able to playback log files on a Taranis transmitter if you have too little memory available.

Log file playback allows you to playback the latest 5 telemetry log files from up to the last 2 weeks on the currently selected model.  You must be _unarmed_ for this to work and if you arm while playing back a log file, the playback will be terminated.

### Setup

Before you can playback a telemetry log file you need to make sure you're logging your telemetry.  By default for each telemetry sensor, the **Logs** option is already checked (unless you unchecked it).  However, it still won't log unless you setup a special function.  In the model setup on your transmitter, page to the **SPECIAL FUNCTIONS** page.  In an empty slot, create a special function that's activated from your arm switch (in my case **SF↓**).  It should look something like this:

```
SF1	SF↓	SD Logs 	0.2s
```

This will log your telemetry 5 times per second automatically every time you arm.  You can set this up on any switch you'd like, but the arm switch is most common.  You can also change the value to 0.1s to log 10 times a second or any other value you wish.  Note that the log playback will be smoother if you select a lower log frequency value.

### Playback

To playback a log file, go to the config menu and scroll to the last item.  If there's any log files in the last 2 weeks, you'll see a date (in YY-MM-DD format).  You can cycle through the dates and select the one you wish to playback (exit will unselect the menu option).

Standard playback will be at close to normal speed (note the timestamp in the title bar).  When playing back, you can use the right stick (mode 2) to fast forward (stick up), reverse (stick down), pause (hold stick right), or quit (stick left).  The further you move the stick up and down the faster the forward or reverse.  The stick center position is playback at normal speed.  When it gets to the end of the log, it will pause.

As multiple flights from the same day are recorded to the same telemetry log files, the playback will continue to the next flight (note the timestamp in the title for time jumps which will be a new flight).


## Horus/Jumper T16 aircraft symbol options

![sample](https://raw.githubusercontent.com/iNavFlight/LuaTelemetry/development/assets/AircraftSymbols.png "Horus aircraft symbols")

CHANGES.md
# Release History

## v1.4.0 - 08/21/2018

* Pilot view (glass cockpit) which includes attitude indicator as well as pilot-familiar layout of additional data
* Shows GPS fix accuracy (HDOP) as strength indicator graph or decimal value (required INAV 2.0.0)
* You can customize when weak GPS fix accuracy (HDOP) triggers alert
* Allows speed sensor selection between GPS speed or pitot sensor's air speed (if available)
* GPS coordinates and altitude are displayed before launch even if there's no GPS fix
* Fuel can report mAh or mWh used instead of percent fuel remaining (percent highly suggested for fuel level alerts)
* The setup, config menu and views loads as separate scripts, greatly reducing memory used
* Variometer graph now correctly uses vertical speed as source instead of Z-axis accelerometer
* Config option under Variometer to also report altitude as voice notifications
* X9D display now uses gray when appropriate to enhance view clarity
* Disable config setting (and set to default) if sensor isn't present
* Config menu wraps from top to bottom (and vice versa) to more easily locate desired config option, also remembers last menu option
* Config menu supports button hold-repeat on X9D and X9D+ to make changes easier
* Config menu fuel percentage warning and critical changed to 1% steps to increase flexibility
* Only shows the last good GPS coordinates instead of the last 5 to reduce complexity (didn't provide more useful info either)
* Removed `getLastPos()` function dependency which is faster, cleans up code and could allow Horus support
* Fixed an issue with "NO TELEM" and "THR WARN" being displayed incorrectly
* Cleaned up readme, broke up into Wiki pages

## v1.3.1 - 05/29/2018

* Support added for the X-Lite transmitter (requires OpenTX v2.2.2)
* Option to set the low and critical warning level for fuel percentage remaining
* Option to display the transmitter voltage as a graph and/or the numerical value
* Modified how battery cell count is calculated
* Can now set cell critical voltage to as low as 2.6v to support 18650 batteries
* Slightly adjusted display positioning to show greater GPS accuracy on QX7 and X-Lite transmitters

## v1.3.0 - 05/13/2018

* Compatibility with new F.Port protocol (requires INAV v1.9.1+)
* Support added for D-series receivers with telemetry (requires INAV v2.0+)
* Option to show GPS coordinates as degrees and minutes or in geocoding format instead of the previous decimal only
* Shows `THR WARN` if not okay to fly and throttle isn't at minimum position
* If distance sensor name is the default `0420`, the max distance now displays correctly
* Updated data digits to display before truncating unit of measure

## v1.2.5 - 04/28/2018

* Uses GPS for launch-based altitude if barometer isn't present
* No longer requires `0420` sensor to be renamed `Dist`
* Better error handling, specially if `iNav` folder doesn't exist
* Renamed PASSTHRU mode MANUAL and add audio file for manual mode
* Correct altitude unit designation in configuration when barometer isn't present
* Adjustment to data display for min/max indicator

## v1.2.4 - 04/17/2018

* Satellite and altitude hold are now icons
* Added flashing notification for headfree mode on QX7 (already existed on X9D)
* Smarter unit designation resulting in a cleaner display (ft = ', MPH & km/h)

## v1.2.3 - 12/19/2017

* Logs the last 5 unique GPS coordinates
* Updated readme screenshots to show variometer
* Release history is now here instead of in readme file

## v1.2.2 - 12/09/2017

* Last 5 GPS coordinates can be reviewed from the config menu
* Resolved issue where if telemetry was lost/recovered it would incorrectly give voice alerts for flight modes/engines armed/disarmed
* Config menu can be accessed at any time, even when armed

## v1.2.1 - 12/01/2017

* Lots of new/changed config settings (press `Menu` button to access) - **Please review!**
* Enhanced and cleaned up config menu, centered menu on QX7 & X9D/+
* Variometer shows if model is gaining or decreasing altitude
* Cell voltage battery view shows with two digits of precision
* Added `S` (South) to X9D/+ directional display

## v1.2.0 - 11/18/2017

* Lua Telemetry is now pre-compiled to greatly reduce memory (source still available)
* Press `Menu` button (when not armed) to modify user configuration options
* Fixed `FAILSAFE`, `RTH`, `NOT OK` & `NO TELEM` modes to flash as they should
* Barometer and magnetometer are now optional (but suggested for full functionality)
* Headfree indication on QX7 changed to show directional indicators as dotted lines
* Startup message/version and error if not running OpenTX v2.2.0 or later

## v1.1.7 - 11/02/2017

* Fix for the default unit type of the `Dist` (`0420`) sensor

## v1.1.6 - 11/01/2017

* On home reset, reset GPS home position, orientation and distance
* Option to display average battery cell voltage instead of total voltage
* Extra digit for data on X9D & X9D+ transmitters
* Variable cleanup saving memory

## v1.1.5 - 10/20/2017

* Voice notification for `Home Reset` with INAV v1.8+
* Moved head free warning on QX7 to top center
* Values convert from decimal to integer when larger to allow for more room
* Better text centering and right justification technique
* Cleaned up code saving more memory

## v1.1.4 - 10/13/2017

* More accurate max altitude alerts and altitude flashes when above max altitude
* Long-press <Enter> resets values (suggest doing this between flights before armed)

## v1.1.3 - 10/10/2017

* Shows metric or imperial values based on transmitter telemetry settings

## v1.1.2 - 10/06/2017

* Lots of refactoring which greatly reduced memory usage
* Re-enabled altitude bar gauge for X9D, X9D+ & X9E transmitters
* Better information layout if no current sensor is present
* Refactored GPS lock calculation to prevent script syntax errors

## v1.1.1 - 09/28/2017

* Refactored code to reduce memory
* Removed altitude bar gauge for X9D, X9D+ & X9E transmitters (used too much memory?)


src/iNav/config.lua
local SMLCD = ...

-- Config options: o=display Order / c=Characters / v=default Value / d=Decimal / x=maX
local config = {
	{ o = 1,  c = 1, v = 1 }, -- Battery View - 1
	{ o = 3,  c = 2, v = 3.5, d = true, x = 3.9}, -- Cell Low - 2
	{ o = 4,  c = 2, v = 3.4, d = true, x = 3.8 }, -- Cell Critical - 3
	{ o = 18, c = 1, v = 2, x = 2 }, -- Voice Alerts - 4
	{ o = 19, c = 1, v = 3, x = 3 }, -- Feedback - 5
	{ o = 11, c = 4, v = -1, x = 9999 }, -- Max Altitude - 6
	{ o = 15, c = 1, v = 0, x = 3 }, -- Variometer - 7
	{ o = 20, c = 1, v = 1 }, -- RTH Feedback - 8
	{ o = 21, c = 1, v = 1 }, -- HeadFree Feedback - 9
	{ o = 22, c = 1, v = 1 }, -- RSSI Feedback - 10
	{ o = 2,  c = 1, v = 2, x = 2 }, -- Battery Alerts - 11
	{ o = 10, c = 1, v = 1 }, -- Altitude Alert - 12
	{ o = 12, c = 1, v = 1, x = 3 }, -- Timer - 13
	{ o = 14, c = 1, v = 1 }, -- Rx Voltage - 14
	{ o = 28, c = 1, v = 0 }, -- HUD Home Icon - 15
	{ o = 33, c = 1, v = 0 }, -- GPS - 16
	{ o = 9,  c = 2, v = 20, x = 40 }, -- Fuel Critical - 17
	{ o = 8,  c = 2, v = 30, x = 50 }, -- Fuel Low - 18
	{ o = 13, c = 1, v = SMLCD and 1 or 2, x = SMLCD and 1 or 2 }, -- Tx Voltage - 19
	{ o = 24, c = 1, v = 0 }, -- Speed Sensor - 20
	{ o = 32, c = 2, v = 3.5, d = true, x = 5.0 }, -- GPS Warning - 21
	{ o = 31, c = 1, v = 0 }, -- GPS HDOP View - 22
	{ o = 6,  c = 1, v = 0, x = 2 }, -- Fuel Unit - 23
	{ o = 16, c = 1, v = 3, x = 9 }, -- Vario Steps - 24
	{ o = 25, c = 1, v = 0, x = 3 }, -- View Mode - 25
	{ o = 23, c = 1, v = 0 }, -- AltHold Center FB - 26
	{ o = 7,  c = 5, v = 1500, x = 9950 }, -- Battery Capacity - 27
	{ o = 17, c = 1, v = 0, x = 6 }, -- Altitude Graph - 28
	{ o = 5,  c = 2, v = 4.3, d = true, x = 4.5 }, -- Cell Calculation - 29
	{ o = 27, c = 1, v = 0, x = 5 }, -- Aircraft Symbol - 30
	{ o = 29, c = 1, v = 0 }, -- Center Map Home - 31
	{ o = 30, c = 1, v = 0 }, -- Orientation - 32
	{ o = 26, c = 1, v = 0 }, -- Roll Scale - 33
	{ o = 34, c = 1, v = 0, l = {[0] = "?"} }, -- Review Log Date - 34
}

for i = 1, #config do
	for ii = 1, #config do
		if i == config[ii].o then
			config[i].z = ii
			config[ii].o = nil
		end
	end
end

return config